configure.ac

```
@@ -1504,13 +1504,54 @@ case $host_os in
         ;;
 esac
 
+# MOLD section
+CUSTOM_LINKER=none
+AC_ARG_ENABLE([mold],
+               AC_HELP_STRING([--enable-mold],
+                              [Use modern linker (mold) to link files]))
+
+if test "x$enable_mold" = "xyes"; then
+   AC_CHECK_PROG([MOLDINSTALLED], [mold], [yes], [no])
+
+   if test "x$MOLDINSTALLED" = "xno"; then
+      AC_MSG_ERROR([`mold` is not installed, you need mold to link with it when using --enable-mold])
+   fi
+
+   AC_PATH_PROG([MOLDPATH], [ld], [no], [/usr/libexec/mold:/usr/local/libexec/mold])
+   if test "x$MOLDPATH" = "xno"; then
+      AC_MSG_ERROR([Unable to find override for ld. Try reinstalling mold])
+   fi
+
+   MOLDPATH="`echo $MOLDPATH | sed 's/\/ld//'`"
```

nit: I think it's cleaner this way:
```suggestion
   MOLDPATH="`dirname "$MOLDPATH"`"
```